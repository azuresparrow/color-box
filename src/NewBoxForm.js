import React, {useState} from "react";
import {v4 as uuid} from "uuid";

const NewBoxForm = ({createRectangle}) => {
    const INITIAL_STATE = {
        height: 10,
        width: 10,
        fill: "#000000"
    }

    const [formData, setFormData] = useState(INITIAL_STATE);
    
    const handleChange = (e) => {
       const {name, value} = e.target;
       setFormData(data => ({
            ...data, 
            [name]:value
       }));
    }
    
    const handleSubmit = (e) => {
        e.preventDefault();
        createRectangle({...formData, id: uuid()});
        setFormData(INITIAL_STATE);
    }
    
    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="width">Width:</label>
            <input id="width" type="number" name="width" placeholder="width" value={formData.width} onChange={handleChange}/>
            <label htmlFor="height">Height:</label>
            <input id="height" type="number"name="height" placeholder="height" value={formData.height} onChange={handleChange}/>
            <label htmlFor="fill">Fill:</label>
            <input id="fill" type="color" name="fill" placeholder="white" value={formData.fill} onChange={handleChange}/>
            <button>Create Box</button>
        </form>
    )
    
}

export default NewBoxForm;