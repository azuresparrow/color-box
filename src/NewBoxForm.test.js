import { render } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

test('renders', () => {
  render(<NewBoxForm />);
});

it('matches snapshot', () => {
  const { asFragment } = render(<NewBoxForm/>);
  expect(asFragment()).toMatchSnapshot();
});