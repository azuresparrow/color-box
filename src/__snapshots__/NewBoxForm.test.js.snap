// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="width"
    >
      Width:
    </label>
    <input
      id="width"
      name="width"
      placeholder="width"
      type="number"
      value="10"
    />
    <label
      for="height"
    >
      Height:
    </label>
    <input
      id="height"
      name="height"
      placeholder="height"
      type="number"
      value="10"
    />
    <label
      for="fill"
    >
      Fill:
    </label>
    <input
      id="fill"
      name="fill"
      placeholder="white"
      type="color"
      value="#000000"
    />
    <button>
      Create Box
    </button>
  </form>
</DocumentFragment>
`;
